### ISSUE TITLE
SvelteKit support?

### ISSUE THREAD

**Issue by jacobbowdoin** [vitebook/vitebook#18]
Do you have any examples of how to use this with a SvelteKit Typescript project?

**Comment by mihar-22** [comment]
Good idea, I should have time to put one together today!

**Comment by tonyrewin** [comment]
`Error: [vite]: Rollup failed to resolve import "$app/stores" from "src/components/Auth.svelte".`  so i've copied svelte preprocessors list and also should manually recreate some aliases or maybe i can find it somewhere in .svelte-kit

**Comment by mihar-22** [comment]
Sorry re-opened, need to fix the alias issue tony mentioned.

**Comment by mihar-22** [comment]
**TLDR;** Vitebook works with SvelteKit now but you can't view application components that are part of SvelteKit which use it's APIs. I recommended splitting your application into components and using props to pass SvelteKit state in. For example, a `<Header />` component that might show links to certain pages should not use the `pages` store directly but rather accept it through a `pages` property. You can then view all your application building blocks (ie: components) in Vitebook, and handle application-level stuff in your SvelteKit environment.

---

This issue is actually tricky depending on the type of SvelteKit support we want. The easy part is viewing isolated Svelte components that don't use any SvelteKit API's (you can use the [SvelteKit example](https://github.com/vitebook/vitebook/tree/main/examples/svelte-kit) I put together to do that). 

The components that do use SvelteKit API's such as routes/pages won't work because it's not a simple matter of configuring some aliases. SvelteKit is a complete solution for building applications and runs on both the client/server. I thought about emulating some of the API's such as the stores for pages, navigation and what not but that still won't resolve other things like the fact that you can have [nested layout files](https://kit.svelte.dev/docs#layouts-nested-layouts), [hooks](https://kit.svelte.dev/docs#hooks) etc.  None of those will work in Vitebook because that's all run inside a SvelteKit application.

To implement true support for all SvelteKit API's, we need to figure out a way to extend Vitebook with the SvelteKit core. It might be possible to implement an interface for how pages/routes are handled in Vitebook so it can delegate that work to a "special" plugin. Essentially we need to abstract out the code responsible for running a dev envinroment in `@vitebook/core`, and to be able to let something like an `AppPlugin` to do the work of resolving routes/pages and other stuff (eg: SvelteKit).

This will clearly be a lot of work and I'm not sure if it's worth it. I feel the solution above in the TLDR should suffice as it's the natural way we build apps.

Closing for now, but let me know if you have any thoughts and I can re-open if needed.

**Comment by jacobbowdoin** [comment]
> You can then view all your application building blocks (ie: components) in Vitebook, and handle application-level stuff in your SvelteKit environment.

Yes, this is how a component mocking/documenting workbench should work so nothing further is needed for my use case. Your simple example was all I needed to get going.

**Comment by iva2k** [comment]
@mihar-22 
> For example, a \<Header /\> component that might show links to certain pages should not use the pages store directly but rather accept it through a pages property. 

It's called "Pure Component". A SvelteKit example app has Header.svelte that uses `pages` store. One can purify that component by moving all actual component stuff into e.g. PureHeader.svelte, and load it in Header.svelte that would pass `pages` stuff down as a property. The story should only show PureHeader.svelte component (since the Header.svelte would be only a wrapper component, without any UI of its own).

BTW, Storybook has exactly the same issue, and they talk about pure components in their docs.
